src/core/modal.slice.ts

```diff
@@ -57,9 +61,9 @@ const modalSlice = createSlice({
           window.history.pushState(
             "",
             "",
-            `?modal=${alreadyOpenModals === null ? "" : alreadyOpenModals}${
-              action.payload.modalId
-            }`
+            appendQueryParametersToUrl(new URL(getCurrentLocation()), {
+              modal: `${alreadyOpenModals ?? ""}${action.payload.modalId}`
+            })
```

Comment:
This is an attempt at not replacing the entire url when opening a modal but instead only adding the new modal.